### PR TITLE
update rolify

### PR DESCRIPTION
**Pull request by differentreality**
It updates rolify and cancancan gems to latest version.

Removes `has_and_belongs_to_many :roles` from **User** model.
Makes necessary changes according to that deletion.

*Note: If we need to find all the users that have role x, we need to execute:
`User.with_role x.name, x.resource`

### CODE REVIEW

File: app/controllers/admin/conference_controller.rb [hunk 1]
```diff
@@ -217,8 +217,8 @@ def conference_params
     def get_users(role_name)
       @role_users = {}
       # Initialize @role variable, so that view can show the role description
-      @role = Role.where(name: role_name, resource: @conference)
-      @role.blank? ? @role_users[role_name] = @role : @role_users[role_name] = @role.first.users
+      @role = Role.where(name: role_name, resource: @conference) || []
```
Comment (hennevogel) [inline app/controllers/admin/conference_controller.rb]: Doe the || trigger on an empty relation? I don't think so.

Comment (differentreality) [inline app/controllers/admin/conference_controller.rb]: Nop, you are right. I was tyring it with `find_by`, when I added the `|| []` part.

File: config/initializers/rolify.rb [hunk 2]
```diff
@@ -4,5 +4,5 @@
 
   # Dynamic shortcuts for User class (user.is_admin? like methods). Default is: false
   # Enable this feature _after_ running rake db:migrate as it relies on the roles table
-  config.use_dynamic_shortcuts
+  # config.use_dynamic_shortcuts
```
Comment (hennevogel) [inline config/initializers/rolify.rb]: What's this disabling? The methods?

File: db/schema.rb [hunk 3]
```diff
@@ -11,307 +11,310 @@
...
 
-ActiveRecord::Schema.define(version: 20160201221411) do
+ActiveRecord::Schema.define(version: 20160226133808) do
 
   create_table "ahoy_events", force: :cascade do |t|
     t.uuid     "visit_id",   limit: 16
-    t.integer  "user_id"
-    t.string   "name"
-    t.text     "properties"
+    t.integer  "user_id",    limit: 4
```
Comment (hennevogel) [inline db/schema.rb]: Hm what causes this change? Sounds strange...

File: db/schema.rb [hunk 4]
```diff
@@ -11,307 +11,310 @@
...
 
   create_table "campaigns", force: :cascade do |t|
-    t.integer  "conference_id"
-    t.string   "name"
-    t.string   "utm_source"
-    t.string   "utm_medium"
-    t.string   "utm_term"
-    t.string   "utm_content"
-    t.string   "utm_campaign"
+    t.integer  "conference_id", limit: 4
```
Comment (hennevogel) [inline db/schema.rb]: Same here. 4? Sounds very limited.

File: config/initializers/rolify.rb (hunk 2 above)
Comment (differentreality) [inline config/initializers/rolify.rb]: Yes, which are some extra methods rolify provides, such as:
user.is_organizer_of? Conference.last

But we are using the simple method:
user.has_role? :organizer, Conference.last

File: db/schema.rb (hunk 3 above)
Comment (hennevogel) [inline db/schema.rb]: Looks like this comes from your DB. Can you recreate it from scratch please?

Comment (hennevogel) [inline db/schema.rb]: @differentreality?

**Comment by differentreality** [comment]
Even after I run `rake db:setup`,I still see the limit for both mysql and sqlite.

**Comment by hennevogel** [comment]
I guess that's a thing then. :eyes:  Thanks for checking

File: Gemfile.lock [hunk 5]
```diff
@@ -334,8 +334,8 @@ GEM
       railties (= 4.2.5.2)
       sprockets-rails
     rails-assets-date.format (1.2.3)
-    rails-assets-holderjs (2.9.1)
-    rails-assets-jquery (2.2.0)
+    rails-assets-holderjs (2.9.3)
```
Comment (sonalkr132) [inline Gemfile.lock]: Why did you update holderjs? [2.9.3 is on rubygems.org](https://rubygems.org/gems/holder_rails/versions/2.9.3) but not on [rails-assets.org](https://rails-assets.org/).  May be we should downgrade for now?

**Pull request merged**